src/pudl/transform/eia930.py

```diff
@@ -0,0 +1,224 @@
+    # after they've been turned into a categorical column with millions of entries
+    # is much faster.
+    netgen_renamed = netgen.rename(
+        lambda col: col.removeprefix("net_generation_").removesuffix("_mw"),
+        axis="columns",
+    ).set_index(qual_cols)
+
+    # Prepare a multi-index for the columns so that we can stack cleanly
+    netgen_renamed.columns = pd.MultiIndex.from_tuples(
+        [x.split("_") for x in netgen_renamed.columns], names=["energy_source", None]
+    )
+    netgen_stacked = (
+        netgen_renamed.stack(level=0, future_stack=True)
+        .rename(columns=lambda x: f"net_generation_{x}_mw")
+        .reset_index()
+        .astype({"energy_source": "string"})
+        .astype(
+            {
+                "balancing_authority_code_eia": pd.CategoricalDtype(),
+                "energy_source": pd.CategoricalDtype(),
+            }
+        )
+    )
+    # TODO[zaneselvans] 2024-04-20: Verify that sum of net generation from all fuels
+    # adds up to the total And then drop the total rows.
+    # NOTE: currently there are some BIG differences between the calculated total and
+    # the reported total.
+
+    # netgen_stacked = netgen_stacked[netgen_stacked["energy_source"] != "total"
+
+    demand = raw_eia930__balance[
+        qual_cols + list(raw_eia930__balance.filter(like="demand"))
+    ].astype({"balancing_authority_code_eia": pd.CategoricalDtype()})
+
+    return (
+        Output(
+            value=netgen_stacked,
+            output_name="core_eia930__hourly_balancing_authority_net_generation",
+        ),
+        Output(
+            value=demand,
+            output_name="core_eia930__hourly_balancing_authority_demand",
+        ),
+    )
+
+
+@multi_asset(
+    outs={
+        "core_eia930__hourly_subregion_demand": AssetOut(
+            io_manager_key="parquet_io_manager"
+        ),
+        "core_eia930__assn_balancing_authority_subregion": AssetOut(
+            io_manager_key="pudl_io_manager"
+        ),
+    },
+    compute_kind="pandas",
+)
+def core_eia930__hourly_subregion_assets(raw_eia930__subregion: pd.DataFrame):
+    """Produce a normalized table of hourly demand by subregion."""
+    demand = (
+        raw_eia930__subregion.assign(
+            subregion_code_eia=lambda df: df["subregion_code_eia"].str.upper()
+        )
+        .astype(
+            {
+                "balancing_authority_code_eia": pd.CategoricalDtype(),
+                "subregion_code_eia": pd.CategoricalDtype(),
+            }
+        )
+        .loc[
+            :,
+            [
+                "report_datetime_local",
+                "report_datetime_utc",
+                "balancing_authority_code_eia",
+                "subregion_code_eia",
+                "demand_reported_mw",
+            ],
+        ]
+    )
+    assn = (
+        demand.groupby("balancing_authority_code_eia")["subregion_code_eia"]
+        .unique()
+        .explode()
+        .to_frame()
+        .reset_index()
```

Comment:
This is kind of a lazy harvesting step.  It just grabs all the values and their associations and assumes they're consistent.  I guess we should check that no subregion is part of more than one BA.

Reply:
i think that would be an easy and valuable little defensive check to add right here

Reply:
Hmm, thinking about this more, I'm not sure this is valid in general. E.g. you could imagine more than one ISO having a subregion `0001` even if they don't at the moment.